GVFS/GVFS/CommandLine/SparseVerb.cs

```diff
@@ -240,23 +240,32 @@ protected override void Execute(GVFSEnlistment enlistment)
 
         private void PruneFoldersOutsideSparse(ITracer tracer, Enlistment enlistment, SparseTable sparseTable)
         {
-            string[] directoriesToDehydrate = this.GetDirectoriesOutsideSparse(enlistment.WorkingDirectoryBackingRoot, sparseTable);
+            string[] directoriesToDehydrate = new string[0];
+            if (!this.ShowStatusWhileRunning(
+                () =>
+                {
+                    directoriesToDehydrate = this.GetDirectoriesOutsideSparse(enlistment.WorkingDirectoryBackingRoot, sparseTable);
+                    return true;
+                },
+                "Finding folders to prune"))
+            {
+                this.ReportErrorAndExit(tracer, "Failed to prune.");
+            }
+
+            this.WriteMessage(tracer, $"Found {directoriesToDehydrate.Length} folders to prune.");
+
             if (directoriesToDehydrate.Length > 0)
             {
-                if (!this.ShowStatusWhileRunning(
-                    () =>
+                ReturnCode verbReturnCode = this.ExecuteGVFSVerb<DehydrateVerb>(
+                    tracer,
+                    verb =>
                     {
-                        ReturnCode verbReturnCode = this.ExecuteGVFSVerb<DehydrateVerb>(
-                            tracer,
-                            verb =>
-                            {
-                                verb.Confirmed = true;
-                                verb.Folders = string.Join(FolderListSeparator, directoriesToDehydrate);
-                            });
-
-                        return verbReturnCode == ReturnCode.Success;
+                        verb.Confirmed = true;
+                        verb.Folders = string.Join(FolderListSeparator, directoriesToDehydrate);

```

Comment:
Alternative to consider for passing in `this.Output`.

You could just set it in the verb configuration action here:

```
verb.Output = this.Output
```

And then `ExecuteGVFSVerb` could be updated to only use the newly created writer if one wasn't set by the configure action:

```
    ReturnCode returnCode;
    returnCode = this.Execute<TVerb>(
    this.EnlistmentRootPathParameter,
    verb =>
    {
        configureVerb?.Invoke(verb);
        verb.Output = verb.Output ?? writer;
    });
```

(It would mean we'd occasionally create a `StringWriter` we don't need, but we've always been creating one so it's not making things any worse)

Reply:
The `GVFSVerb.Output` gets set in the constructor so I don't think it will every be null unless we make more changes and not set it in the constructor but I would need to see if that would have any bad side effects.
```
        public GVFSVerb(bool validateOrigin = true)
        {
            this.Output = Console.Out;
```

Reply:
I totally missed that, thanks!